src/watch_filter.rs

```diff
@@ -173,11 +200,68 @@ impl Filterer for WatchFilter {
         }
 
         // By process of elimination the event is valid
-        tracing::trace!("passing event: {event:?}");
+        tracing::debug!("passing event: {event:?}");
         Ok(true)
     }
 }
 
+#[derive(Debug)]
+pub struct Config {
+    pub manifest_pattern: WatchPattern,
+    pub source_patterns: Vec<WatchPattern>,
+    pub artifact_patterns: Vec<WatchPattern>,
+    pub ignore_patterns: Vec<WatchPattern>,
+}
+
+pub struct WatchPattern {
+    pub active: bool,
+    pub glob: String,
+    pub pattern: Pattern,
+}
+
+impl fmt::Debug for WatchPattern {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "WatchPattern {{ glob: {:?}, active: {:?} }}",
+            self.glob, self.active
+        )
```

Comment:
This is what `#[derive(Debug)]` does internally (minus the one field):

```suggestion
        f.debug_struct("WatchPattern")
            .field("active", &self.active)
            .field("glob", &self.glob)
            .finish()
```

Reply:
Thanks. Keep the hits coming 😄 . I'm learning a lot about Rust in these code reviews.